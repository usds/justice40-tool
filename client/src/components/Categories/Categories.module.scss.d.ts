declare namespace CategoriesNamespace {
    export interface ICategoriesScss {
        categoriesContainer: string;
        tribalLandsPara: string;
    }
  }

declare const CategoriesScssModule: CategoriesNamespace.ICategoriesScss & {
    /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
    locals: CategoriesNamespace.ICategoriesScss;
  };

  export = CategoriesScssModule;
