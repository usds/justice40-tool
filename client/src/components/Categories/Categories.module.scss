@use '../../styles/design-system.scss' as *;

.categoriesContainer {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  @include u-margin-top(4);
  
  .tribalLandsPara {
    @include u-margin-bottom(6);
    
  }
}
