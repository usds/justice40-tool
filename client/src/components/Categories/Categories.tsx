import React from 'react';
import {useIntl} from 'gatsby-plugin-intl';
import {Grid} from '@trussworks/react-uswds';

import CategoryCard from '../CategoryCard';
import J40MainGridContainer from '../J40MainGridContainer';

import * as METHODOLOGY_COPY from '../../data/copy/methodology';
import * as styles from './Categories.module.scss';

const categories = [
  METHODOLOGY_COPY.CATEGORIES.CLIMATE_CHANGE,
  METHODOLOGY_COPY.CATEGORIES.CLEAN_ENERGY,
  METHODOLOGY_COPY.CATEGORIES.HEALTH_BURDENS,
  METHODOLOGY_COPY.CATEGORIES.AFFORDABLE_HOUSING,
  METHODOLOGY_COPY.CATEGORIES.LEGACY_POLLUTION,
  METHODOLOGY_COPY.CATEGORIES.CLEAN_TRANSPORT,
  METHODOLOGY_COPY.CATEGORIES.CLEAN_WATER,
  METHODOLOGY_COPY.CATEGORIES.WORKFORCE_DEV,
];

const Categories = () => {
  const intl = useIntl();

  return (
    <>
      <J40MainGridContainer>

        <Grid row>
          <Grid col={8}>
            <h2>{METHODOLOGY_COPY.CATEGORY.HEADING}</h2>
            <p>{intl.formatMessage(METHODOLOGY_COPY.PAGE.PARA3)}</p>
            <p>{intl.formatMessage(METHODOLOGY_COPY.PAGE.PARA4)}</p>
            <p>{METHODOLOGY_COPY.FORMULA.PARA6}</p>
          </Grid>
        </Grid>

      </J40MainGridContainer>

      <J40MainGridContainer className={styles.categoriesContainer}>
        {
          categories.map((category, index) => <CategoryCard key={index} categoryInfo={category} />)
        }
      </J40MainGridContainer>

      <J40MainGridContainer>

        <Grid row className={styles.tribalLandsPara}>
          <Grid col={8}>
            <h2>{intl.formatMessage(METHODOLOGY_COPY.PAGE.SUB_HEADING_2)}</h2>
            <p>{intl.formatMessage(METHODOLOGY_COPY.PAGE.PARA5)}</p>
          </Grid>
        </Grid>

      </J40MainGridContainer>

    </>
  );
};

export default Categories;
