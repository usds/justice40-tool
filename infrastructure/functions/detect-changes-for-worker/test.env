REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1