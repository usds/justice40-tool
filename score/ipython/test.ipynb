{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
