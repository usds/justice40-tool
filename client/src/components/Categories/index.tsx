import Categories from './Categories';

export default Categories;
