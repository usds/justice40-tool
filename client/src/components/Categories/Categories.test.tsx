import * as React from 'react';
import {render} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import Categories from './Categories';

describe('rendering of the Categories', () => {
  const {asFragment} = render(
      <LocalizedComponent>
        <Categories />
      </LocalizedComponent>,
  );

  it('checks if component renders', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
