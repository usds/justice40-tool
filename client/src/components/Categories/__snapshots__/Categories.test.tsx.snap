// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of the Categories checks if component renders 1`] = `
<DocumentFragment>
  <div
    class="grid-container-desktop-lg"
    data-testid="gridContainer"
  >
    <div
      class="grid-row"
      data-testid="grid"
    >
      <div
        class="grid-col-8"
        data-testid="grid"
      >
        <h2>
          Categories of Burdens
        </h2>
        <p>
          
      The tool uses datasets as indicators of burdens. The burdens are organized into categories. A community is highlighted as disadvantaged on the CEJST map if it is in a census tract that is (1) at or above the threshold for one or more environmental, climate, or other burdens, and (2) at or
      above the threshold for an associated socioeconomic burden.
    
        </p>
        <p>
          
      In addition, a census tract that is completely surrounded by disadvantaged communities and is at or above the 50% percentile for low income is also considered disadvantaged.
    
        </p>
        <p>
          
      Census tracts are small units of geography. Census tract boundaries for 
          <a
            class="usa-link usa-link--external"
            data-cy=""
            href="https://www.census.gov/programs-surveys/geography/about/glossary.html#par_textimage_6"
            rel="noreferrer"
            target="_blank"
          >
            statistical areas
          </a>
           are determined by the U.S. Census Bureau once every ten years. The tool utilizes the census tract boundaries from 2010. This was chosen because many of the data sources in the tool currently use the 2010 
      census boundaries.    
    
        </p>
      </div>
    </div>
  </div>
  <div
    class="grid-container-desktop-lg"
    data-testid="gridContainer"
  >
    <div>
      <h3>
        Climate change
      </h3>
      <p>
        
      Communities are 
        <strong>
          identified as disadvantaged
        </strong>
         if they are in census tracts that:
    
      </p>
      <p>
        <strong>
          ARE
        </strong>
         at or above the 90th percentile for 
        <a
          class="usa-link"
          href="#exp-agr-loss-rate"
        >
          expected agriculture loss rate
        </a>
         OR 
        <a
          class="usa-link"
          href="#exp-bld-loss-rate"
        >
          expected building loss rate
        </a>
         OR 
        <a
          class="usa-link"
          href="#exp-pop-loss-rate"
        >
          expected population loss rate
        </a>
          OR 
        <a
          class="usa-link"
          href="#flood-risk"
        >
          projected flood risk
        </a>
         OR 
        <a
          class="usa-link"
          href="#wildfire-risk"
        >
          projected wildfire risk
        </a>
      </p>
      <p>
        
      
        <strong>
          AND
        </strong>
         are at or above the 65th percentile for 
        <a
          class="usa-link"
          href="#low-income"
        >
          low income
        </a>
        
    
      </p>
    </div>
    <div>
      <h3>
        Energy
      </h3>
      <p>
        
      Communities are 
        <strong>
          identified as disadvantaged
        </strong>
         if they are in census tracts that:
    
      </p>
      <p>
        <strong>
          ARE
        </strong>
         at or above the 90th percentile for 
        <a
          class="usa-link"
          href="#energy-burden"
        >
          energy cost
        </a>
         OR 
        <a
          class="usa-link"
          href="#pm-25"
        >
          PM2.5 in the air
        </a>
      </p>
      <p>
        
      
        <strong>
          AND
        </strong>
         are at or above the 65th percentile for 
        <a
          class="usa-link"
          href="#low-income"
        >
          low income
        </a>
        
    
      </p>
    </div>
    <div>
      <h3>
        Health
      </h3>
      <p>
        
      Communities are 
        <strong>
          identified as disadvantaged
        </strong>
         if they are in census tracts that:
    
      </p>
      <p>
        <strong>
          ARE
        </strong>
         at or above the 90th percentile for 
        <a
          class="usa-link"
          href="#asthma"
        >
          asthma
        </a>
         OR 
        <a
          class="usa-link"
          href="#diabetes"
        >
          diabetes
        </a>
         OR 
        <a
          class="usa-link"
          href="#heart-disease"
        >
          heart disease
        </a>
         OR 
        <a
          class="usa-link"
          href="#life-exp"
        >
          low life expectancy
        </a>
      </p>
      <p>
        
      
        <strong>
          AND
        </strong>
         are at or above the 65th percentile for 
        <a
          class="usa-link"
          href="#low-income"
        >
          low income
        </a>
        
    
      </p>
    </div>
    <div>
      <h3>
        Housing
      </h3>
      <p>
        
      Communities are 
        <strong>
          identified as disadvantaged
        </strong>
         if they are in census tracts that:
    
      </p>
      <p>
        Experienced 
        <a
          class="usa-link"
          href="#hist-underinv"
        >
          historic underinvestment
        </a>
         OR are at or above the 90th percentile for the 
        <a
          class="usa-link"
          href="#house-burden"
        >
          housing cost
        </a>
         OR 
        <a
          class="usa-link"
          href="#green-space"
        >
          lack of green space
        </a>
         OR 
        <a
          class="usa-link"
          href="#indoor-plumb"
        >
          lack of indoor plumbing
        </a>
         OR 
        <a
          class="usa-link"
          href="#lead-paint"
        >
          lead paint
        </a>
      </p>
      <p>
        
      
        <strong>
          AND
        </strong>
         are at or above the 65th percentile for 
        <a
          class="usa-link"
          href="#low-income"
        >
          low income
        </a>
        
    
      </p>
    </div>
    <div>
      <h3>
        Legacy pollution
      </h3>
      <p>
        
      Communities are 
        <strong>
          identified as disadvantaged
        </strong>
         if they are in census tracts that:
    
      </p>
      <p>
        Have at least one 
        <a
          class="usa-link"
          href="#mine-land"
        >
          abandoned mine land
        </a>
         OR 
        <a
          class="usa-link"
          href="#fuds"
        >
          Formerly Used Defense Sites
        </a>
         OR are at or above the 90th percentile for 
        <a
          class="usa-link"
          href="#prox-haz"
        >
          proximity to hazardous waste facilities
        </a>
         OR 
        <a
          class="usa-link"
          href="#prox-npl"
        >
          proximity to Superfund sites (National Priorities List (NPL))
        </a>
         OR 
        <a
          class="usa-link"
          href="#prox-rmp"
        >
          proximity to Risk Management Plan (RMP) facilities
        </a>
      </p>
      <p>
        
      
        <strong>
          AND
        </strong>
         are at or above the 65th percentile for 
        <a
          class="usa-link"
          href="#low-income"
        >
          low income
        </a>
        
    
      </p>
    </div>
    <div>
      <h3>
        Transportation
      </h3>
      <p>
        
      Communities are 
        <strong>
          identified as disadvantaged
        </strong>
         if they are in census tracts that:
    
      </p>
      <p>
        
        
        <strong>
          ARE
        </strong>
         at or above the 90th percentile for 
        <a
          class="usa-link"
          href="#diesel-pm"
        >
          diesel particulate matter exposure
        </a>
         OR 
        <a
          class="usa-link"
          href="#trans-barrier"
        >
          transportation barriers
        </a>
         OR 
        <a
          class="usa-link"
          href="#traffic-vol"
        >
          traffic proximity and volume
        </a>
        
      
      </p>
      <p>
        
      
        <strong>
          AND
        </strong>
         are at or above the 65th percentile for 
        <a
          class="usa-link"
          href="#low-income"
        >
          low income
        </a>
        
    
      </p>
    </div>
    <div>
      <h3>
        Water and wastewater
      </h3>
      <p>
        
      Communities are 
        <strong>
          identified as disadvantaged
        </strong>
         if they are in census tracts that:
    
      </p>
      <p>
        <strong>
          ARE
        </strong>
         at or above the 90th percentile for 
        <a
          class="usa-link"
          href="#leaky-uwt"
        >
          underground storage tanks and releases
        </a>
         OR 
        <a
          class="usa-link"
          href="#waste-water"
        >
          wastewater discharge
        </a>
      </p>
      <p>
        
      
        <strong>
          AND
        </strong>
         are at or above the 65th percentile for 
        <a
          class="usa-link"
          href="#low-income"
        >
          low income
        </a>
        
    
      </p>
    </div>
    <div>
      <h3>
        Workforce development
      </h3>
      <p>
        
      Communities are 
        <strong>
          identified as disadvantaged
        </strong>
         if they are in census tracts that:
    
      </p>
      <p>
        
        
        <strong>
          ARE
        </strong>
         at or above the 90th percentile for 
        <a
          class="usa-link"
          href="#ling-iso"
        >
          linguistic isolation
        </a>
         OR 
        <a
          class="usa-link"
          href="#low-med-inc"
        >
          low median income
        </a>
         OR 
        <a
          class="usa-link"
          href="#poverty"
        >
          poverty
        </a>
         OR 
        <a
          class="usa-link"
          href="#unemploy"
        >
          unemployment
        </a>
        
      
      </p>
      <p>
        <strong>
          AND
        </strong>
         fewer than 10% of people ages 25 or older have a 
        <a
          class="usa-link"
          href="#high-school"
        >
          high school education
        </a>
         (i.e. graduated with a high school diploma)
  
      </p>
    </div>
  </div>
  <div
    class="grid-container-desktop-lg"
    data-testid="gridContainer"
  >
    <div
      class="grid-row"
      data-testid="grid"
    >
      <div
        class="grid-col-8"
        data-testid="grid"
      >
        <h2>
          
      Tribes
    
        </h2>
        <p>
          
      Federally Recognized Tribes, including Alaska Native Villages, are also considered disadvantaged communities.
    
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
